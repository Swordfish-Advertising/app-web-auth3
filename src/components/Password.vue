<template>
  <div>
    <v-text-field
      label="Password"
      v-model="password"
      :append-icon="visiblePass ? 'lock_open' : 'lock'"
      @click:append="() => (visiblePass = !visiblePass)"
      :type="visiblePass ? 'text' : 'password'"
      :rules="[rules.required]"
    ></v-text-field>
    <v-text-field
      label="Password confirmation"
      v-if="confirmation"
      v-model="repass"
      :append-icon="visibleRepass ? 'lock_open' : 'lock'"
      @click:append="() => (visibleRepass = !visibleRepass)"
      :type="visibleRepass ? 'text' : 'password'"
      :rules="[matchPassword]"
    ></v-text-field>
  </div>
</template>

<script>
  export default {
    props: ['confirmation'],
    data: () => ({
      visiblePass: false,
      visibleRepass: false,
      password: '',
      repass: '',
      rules: {
        required: value => !!value || 'Password is required.'
      }
    }),
    computed: {
      matchPassword: function () {
        return this.password === this.repass || 'Password confirmation does not match.'
      }
    }
  }
</script>