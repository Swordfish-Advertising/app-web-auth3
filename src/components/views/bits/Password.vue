<template>
  <div>
    <v-text-field
      id="password"
      label="Password"
      v-bind:value="value"
      v-on:input="$emit('input', $event)"
      :append-icon="visiblePass ? 'lock_open' : 'lock'"
      @click:append="() => (visiblePass = !visiblePass)"
      :type="visiblePass ? 'text' : 'password'"
      :rules="[rules.required]"
    ></v-text-field>
    <v-text-field
      id="passConfirmation"
      label="Password confirmation"
      v-if="confirmation"
      v-model="repass"
      :type="visiblePass ? 'text' : 'password'"
      :rules="[matchPassword]"
    ></v-text-field>
  </div>
</template>

<script>
  export default {
    props: ['confirmation', 'value'],
    data: () => ({
      visiblePass: false,
      repass: '',
      rules: {
        required: value => !!value || 'Password is required.'
      }
    }),
    computed: {
      matchPassword: function () {
        return this.value === this.repass || 'Password confirmation does not match.'
      }
    }
  }
</script>