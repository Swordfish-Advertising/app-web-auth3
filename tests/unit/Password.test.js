/* global describe, it, beforeEach, expect */

import { shallowMount } from '@vue/test-utils';
import Password from '../../src/components/Password';
import Vue from 'vue';
import Vuetify from 'vuetify';

Vue.use(Vuetify);

describe('RegisterUser.test.js', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(Password);
  });

  it('renders correctly (snapshots matching)', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('asks only for password by default', () => {
    const password = wrapper.find('#password');
    expect(password.exists()).toBe(true);
    const confirmation = wrapper.find('#passConfirmation');
    expect(confirmation.exists()).toBe(false);
  });

  it('asks for password and confirmation if needed', () => {
    wrapper = shallowMount(Password, {
      propsData: {
        confirmation: true
      }
    });
    const password = wrapper.find('#password');
    expect(password.exists()).toBe(true);
    const confirmation = wrapper.find('#passConfirmation');
    expect(confirmation.exists()).toBe(true);
  });

  it('validates password correctly', () => {
    const requiredRule = wrapper.vm.rules.required;
    expect(requiredRule()).toBe('Password is required.');
    expect(requiredRule('notEmpty')).toBe(true);
  });

  it('validates password confirmation correctly', () => {
    wrapper.setData({
      password: '1234',
      repass: '1234'
    });
    expect(wrapper.vm.matchPassword).toBe(true);
    wrapper.setData({
      password: '1234',
      repass: '4321'
    });
    expect(wrapper.vm.matchPassword).toBe('Password confirmation does not match.');
  });
});
