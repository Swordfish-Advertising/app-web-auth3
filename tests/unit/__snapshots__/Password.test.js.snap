// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password.test.js renders correctly (snapshots matching) 1`] = `
<div>
  <vtextfield-stub
    append-icon="lock"
    id="password"
    label="Password"
    rules="function required(value) {
          return !!value || 'Password is required.';
        }"
    type="password"
  />
   
  <!---->
</div>
`;
